packages/gatsby-source-contentful/src/extend-node-type.js

```diff
@@ -98,19 +102,25 @@ const getBase64Image = imageProps => {
     })
   }
 
-  const promise = new Promise((resolve, reject) => {
-    base64Img.requestBase64(requestUrl, (a, b, body) => {
-      // TODO: against dogma, confirm whether writeFileSync is indeed slower
-      fs.promises
-        .writeFile(cacheFile, body)
-        .then(() => resolve(body))
-        .catch(e => {
-          console.error(
-            `Contentful:getBase64Image: failed to write ${body.length} bytes remotely fetched from \`${requestUrl}\` to: \`${cacheFile}\`\nError: ${e}`
-          )
-          reject(e)
-        })
+  const promise = new Promise(async (resolve, reject) => {
+    const imageResponse = await axios.get(requestUrl, {
+      responseType: `arraybuffer`,
     })
+
+    const base64 = Buffer.from(imageResponse.data, `binary`).toString(`base64`)
+
+    const body = `data:image/jpeg;base64,${base64}`
+
+    // TODO: against dogma, confirm whether writeFileSync is indeed slower
+    fs.promises
```

Comment:
You have fs-extra in the project which promisifies all of the fs calls, so you can just use that.

Reply:
But why go fs-extra if we have it supported in the language?

Reply:
It is marked as stable api:

https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_fs_promises_api